lib/tasks/assets.rake

```diff
@@ -1,14 +1,4 @@
 namespace :assets do
-  desc "Mark an asset as deleted"
-  task :delete, [:id] => :environment do |_t, args|
-    Asset.find(args.fetch(:id)).destroy
-  end
-
-  desc "Restore an asset after being deleted"
-  task :restore, [:id] => :environment do |_t, args|
-    Asset.find(args.fetch(:id)).restore
-  end
```

Comment:
Do we ever use these? I can't see a justification for it in the commits where this was added, but I wonder if a 'soft delete' may be a desirable feature in some situations?

Reply:
I believe we added soft-delete tasks because we wanted that to be the default unless otherwise specified. For example, if an asset needs to be removed because it's outdated and still showing up in Google, we would do a soft delete. If an asset needs to be removed because it affects national security, then it would be hard deleted.

Reply:
That's interesting. Can you help me understand the distinction better? I believe we only run these tasks when an asset has been orphaned. What's the risk of always doing a hard delete?

Reply:
Well, the risk is that if we needed to restore an asset for whatever reason or get access to an asset that's not live on the site anymore. The normal behaviour of Asset Manager is not to delete the files from S3 when they get replaced or removed from documents (only [one search result for `cloud_storage.delete`](https://github.com/alphagov/asset-manager/search?q=cloud_storage.delete&unscoped_q=cloud_storage.delete)) so this matches that behaviour.

Reply:
Ah, OK. That's quite an obscure distinction in [the docs](https://docs.publishing.service.gov.uk/manual/howto-manually-remove-assets.html) - a choice is given, but without a way to make it. If we do support both kinds of behaviour, then we should indicate when to use them.

I've added another commit to make soft deletion the default. However, I don't think we need a task for restoring assets, since it's not a common or documented thing to do.

I can also update the docs with some guidance between the two tasks.

How does that sound?

Reply:
Sounds reasonable to me.

As a side point, maybe it's just me, but I would consider the Rake tasks themselves to be documentation. `rake -T` can be used to list the tasks, and that prints useful information like the description and the arguments. Following on from that, I would suggest improving the documentation of the Rake tasks themselves rather than having something in the dev docs which is far more likely to end up out of date with the specifics in the task itself.

Reply:
While I would normally agree, the problem is that these tasks are referred to from dev docs as part of the overall process of removing an asset. The links in the docs go straight to Jenkins, so there isn't an opportunity to run `rake -T`. Since we already have docs about removing an asset, which extend beyond asset manager, it makes sense to explain the distinction there, since that's probably where we will be looking to understand it.